visidata/pyobj.py

```diff
@@ -3,16 +3,18 @@
 from visidata import *
 
 option('visibility', 0, 'visibility level')
+option('expand_col_scanrows', 1000, 'number of rows to check when expanding columns (0 = all)')
 
 
 class PythonSheet(Sheet):
     def openRow(self, row):
         return load_pyobj("%s[%s]" % (self.name, self.keystr(row)), row)
 
 
-def expand_cols_deep(sheet, cols, rows, depth=0):  # depth == 0 means drill all the way
+def expand_cols_deep(sheet, cols, rows=None, depth=0):  # depth == 0 means drill all the way
     'expand all visible columns of containers to the given depth (0=fully)'
     ret = []
+    rows = rows or sheet.rows[:options.expand_col_scanrows] or sheet.rows
```

Comment:
One issue with this, is that if you find a larger list far down a large dataset, you won't be able to expand to use it, because it'll always use the same first 1000 rows.  I don't know if it's worth it to implement, but I guess the ideal would be "1000 rows around the cursor row" or something like that.

Reply:
Ah that's an interesting case. I like that suggestion! I considered and rejected using a random sample of rows to avoid overly weighting the top rows, but that seemed really evil :). Using the cursor row as guidance seems like a smarter approach, thanks!

Reply:
Ok, I made a little change so instead of "first n rows" it will use "n rows centered around the cursor row" as you suggested. So for datasets larger than 1k rows, the default behavior (assuming you expand before moving) would look at the first and last 500.

At this point I'm still not sure if it would be safer to default to 1,000 and have people tweak to be more thorough, or default to 0 (all rows) and have people tweak for speed. I left 1,000 for now but would be fine flipping it.

Reply:
Cool, thanks AJ!  I think I like the default to be 1000 since it covers 99% of cases, including the big and the small and the usual.  It only doesn't cover the "extra field not found in 1000 rows around cursor" case, which we should mention in the docs for these commands.

Reply:
Sounds good to me. When you say "the docs for these commands", where would that be? There's a mention of expanding columns in the json section of the man page, but I'm not sure if there's a separate command help somewhere that I'm missing.